client/v3/config.go

```diff
@@ -118,3 +119,56 @@ type AuthConfig struct {
 	Username string `json:"username"`
 	Password string `json:"password"`
 }
+
+// NewClientConfig creates a Config based on the provided ConfigSpec.
+func NewClientConfig(confSpec *ConfigSpec, lg *zap.Logger) (*Config, error) {
+	var cfgtls *transport.TLSInfo
+
+	if confSpec.Secure != nil {
```

Comment:
I think this code can be further refactored by extracting `func newTLSConfig(cfg *SecureConfig, lg *zap.Logger) (*tls.Config, error)`.

Reply:
makes sense.

Reply:
done.